File: SoloLearn Codes/Camel to Snake/CamelToSnake.cs
using System;
using System.Linq;
using System.Text.RegularExpressions;

namespace SoloLearn
{
    class Program
    {
        static void Main(string[] args)
        {
            string s = Console.ReadLine();
            Console.WriteLine(Regex.Replace(string.Join("", s.Select(c => c.ToString()).Select(c => c == c.ToUpper() ? "_" + c : c).ToArray()).ToLower(), @"^_+", ""));
        }
    }
}

File: SoloLearn Codes/Camel to Snake/camel_to_snake.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {

    string s;
    getline(cin, s);

    for (int i = 0; i < s.length(); i++) {
        if (s[i] == toupper(s[i]) && i != 0) cout << "_";
        cout << (char) tolower(s[i]);
    }

    return 0;
}